modules/node/cms_node_api.e

```diff
@@ -249,6 +249,26 @@ feature -- Access: Node
 			end
 		end
 
+	user_is_node_owner (u: READABLE_STRING_32; nid: INTEGER_64): BOOLEAN
+			-- Is the user `u' owner of the node `n'.
+		do
+			if attached {CMS_USER} node_storage.node_author (nid) as l_user then
+				Result := l_user.name.is_case_insensitive_equal (u)
+			end
+		end
+
+feature -- Permission Scope: Node
+
+	permission_scope (u: detachable READABLE_STRING_32; nid: INTEGER_64): STRING
+			-- Result 'own' if the user `u' is the owner of the node `nid', in other case
+			-- `any'.
+		do
+			Result := "any"
```

Comment:
What if a role has   "delete any page" enabled, and "delete own page" disabled.
this would be an error made by the maintainer while configuring the permissions.

But this may occur.
So I don't know if this  "any" or "own" is good or not.